lighthouse-core/test/fraggle-rock/gather/runner-helpers-test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const helpers = require('../../../fraggle-rock/gather/runner-helpers.js');
+const Gatherer = require('../../../fraggle-rock/gather/base-gatherer.js');
+
+/* eslint-env jest */
+
+describe('collectArtifactDependencies', () => {
+  /** @type {LH.Config.ArtifactDefn} */
+  let artifact;
+  /** @type {Record<string, any>} */
+  let artifactsById;
+
+  beforeEach(() => {
+    artifact = {
+      id: 'Artifact',
+      gatherer: {instance: new Gatherer()},
+      dependencies: {ImageElements: {id: 'Dependency'}},
+    };
+    artifactsById = {
+      Dependency: [],
+    };
+  });
+
+  it('should handle no dependencies', async () => {
```

Comment:
Would this be the same as `dependencies: {}`?

Reply:
I'm not sure `dependencies: {}` is a valid type given the way I've setup the constraints, but it is semantically equivalent, yes.

If it is an invalid type, are you thinking a `// @ts-expect-error` test anyway?

Reply:
added one for plugin case 👍